plugins/askrene/explain_failure.c

```diff
@@ -0,0 +1,290 @@
+				      fmt_amount_msat(tmpctx,
+						      amount_msat(fp16_to_u64(htlc_max))));
+	}
+
+	/* This seems unlikely, but don't return NULL. */
+	if (!ret)
+		ret = tal_fmt(ctx, "is constrained");
+	return ret;
+}
+
+struct stat {
+	size_t num_channels;
+	struct amount_msat capacity;
+};
+
+struct node_stats {
+	struct stat total, gossip_known, enabled;
+};
+
+enum node_direction {
+	INTO_NODE,
+	OUT_OF_NODE,
+};
+
+static void add_stat(struct stat *stat,
+		     struct amount_msat amount)
+{
+	stat->num_channels++;
+	if (!amount_msat_accumulate(&stat->capacity, amount))
+		abort();
+}
+
+static void node_stats(const struct route_query *rq,
+		       const struct gossmap_node *node,
+		       enum node_direction node_direction,
+		       struct node_stats *stats)
+{
+	memset(stats, 0, sizeof(*stats));
+	for (size_t i = 0; i < node->num_chans; i++) {
+		int dir;
+		struct gossmap_chan *c;
+		struct amount_msat cap_msat;
+
+		c = gossmap_nth_chan(rq->gossmap, node, i, &dir);
+		cap_msat = gossmap_chan_get_capacity(rq->gossmap, c);
+
+		if (node_direction == INTO_NODE)
+			dir = !dir;
+
+		add_stat(&stats->total, cap_msat);
+		if (gossmap_chan_set(c, dir))
+			add_stat(&stats->gossip_known, cap_msat);
+		if (c->half[dir].enabled)
+			add_stat(&stats->enabled, cap_msat);
+	}
+}
+
+static const char *check_capacity(const tal_t *ctx,
+				  const struct route_query *rq,
+				  const struct gossmap_node *node,
+				  enum node_direction node_direction,
+				  struct amount_msat amount,
+				  const char *name)
+{
+	struct node_stats stats;
+
+	node_stats(rq, node, node_direction, &stats);
+	if (amount_msat_greater(amount, stats.total.capacity)) {
+		return tal_fmt(ctx,
+			       NO_USABLE_PATHS_STRING
+			       "  Total %s capacity is only %s"
+			       " (in %zu channels).",
+			       name,
+			       fmt_amount_msat(tmpctx, stats.total.capacity),
+			       stats.total.num_channels);
+	}
+	if (amount_msat_greater(amount, stats.gossip_known.capacity)) {
+		return tal_fmt(ctx,
+			       NO_USABLE_PATHS_STRING
+			       "  Missing gossip for %s: only known %zu/%zu channels, leaving capacity only %s of %s.",
+			       name,
+			       stats.gossip_known.num_channels,
+			       stats.total.num_channels,
+			       fmt_amount_msat(tmpctx, stats.gossip_known.capacity),
+			       fmt_amount_msat(tmpctx, stats.total.capacity));
+	}
+	if (amount_msat_greater(amount, stats.enabled.capacity)) {
+		return tal_fmt(ctx,
+			       NO_USABLE_PATHS_STRING
+			       "  The %s has disabled %zu of %zu channels, leaving capacity only %s of %s.",
+			       name,
+			       stats.total.num_channels - stats.enabled.num_channels,
+			       stats.total.num_channels,
+			       fmt_amount_msat(tmpctx, stats.enabled.capacity),
+			       fmt_amount_msat(tmpctx, stats.total.capacity));
+	}
+	return NULL;
+}
+
+/* Return description of why scidd is disabled scidd */
+static const char *describe_disabled(const tal_t *ctx,
```

Comment:
There is another case when a channel is disabled because it is connected to a disabled node.